### PR TITLE
bump sigstore-protobuf-specs

### PR DESCRIPTION
**Pull request by woodruffw**
Bumps us to `~= 0.3.2`. This patch release enables type hinting, allowing us to refine the types a bit here.

### CODE REVIEW

File: sigstore/dsse.py [hunk 1]
```diff
@@ -217,7 +216,7 @@ def _sign(key: ec.EllipticCurvePrivateKey, stmt: Statement) -> Envelope:
         _Envelope(
             payload=stmt._contents,
             payload_type=Envelope._TYPE,
-            signatures=[Signature(sig=signature, keyid=None)],
+            signatures=[Signature(sig=signature)],
```
Comment (jku) [inline sigstore/dsse.py:219]: How does this work if keyid is not marked optional in the protobuf -- am I missing something?

**Review by jku — APPROVED** [review]
If this works, great. If you can explain how Signature works without keyid, that's appreciated.

**Comment by woodruffw** [comment]
> If you can explain how Signature works without keyid, that's appreciated.

From looking at the proto definition, the keyid field is entirely optional. But the code generator isn't smart enough to understand that optional means that the kwarg is also nullable, so I just removed it outright instead of explicitly setting it to `None` 😅

TL;DR: The generated message is completely unchanged, we're just correctly handling an overly strict type hint now.
